src/Model.php

```diff
@@ -441,7 +441,7 @@ public function toJson($options = 0)
      *
      * @return array
      */
-    public function jsonSerialize()
+    public function jsonSerialize(): array
```

Comment:
Instead of changing the hard types now, a temporary attribute can be applied for PHP 8.0 & 8.1 that suppresses deprecation notices: https://wiki.php.net/rfc/internal_method_return_types

```php
     *
     * @return array
     */
    #[ReturnTypeWillChange]
    public function jsonSerialize()
    {
```

This is the approach "laravel/framework" (and all its other packages) took to transition to the core's new method signatures.

PHP 9.0 will require an exact method signature like `public function jsonSerialize(): array` so the whole PHP ecosystem will then require new major versions for all Composer packages.

`#[ReturnTypeWillChange]` will allow this PR to be merged for a minor version release and the compatibility pain can be delayed for later.